### ISSUE TITLE
looping on rtmidi.get_output_names() gives a crash

### ISSUE THREAD

**Issue by folkertvanheusden** [mido/mido#330]
Hi,

I tried calling rtmidi.get_output_names() in a loop to automatically see new devices appear.
This works (also for input) only it crashes after a bit (also for input):

```
folkert@lappiemctopface:~/Projects/RJRalsa$ python3 test.py 
['Client-129:RtMidi Input 129:0', 'FLUID Synth (50424):Synth input port (50424:0) 161:0', 'Midi Through:Midi Through Port-0 14:0', 'aseqdump:aseqdump 132:0', 'rtpmidi lappiemctopface:Network 128:0', 'rtpmidi lappiemctopface:cameras4 128:1']
...
['Client-129:RtMidi Input 129:0', 'FLUID Synth (50432):Synth input port (50432:0) 171:0', 'Midi Through:Midi Through Port-0 14:0', 'aseqdump:aseqdump 132:0', 'rtpmidi lappiemctopface:Network 128:0', 'rtpmidi lappiemctopface:cameras4 128:1']
ALSA lib seq_hw.c:466:(snd_seq_hw_open) open /dev/snd/seq failed: Cannot allocate memory
Traceback (most recent call last):
  File "src/_rtmidi.pyx", line 815, in rtmidi._rtmidi.MidiIn.__cinit__
RuntimeError: MidiInAlsa::initialize: error creating ALSA sequencer client object.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/folkert/Projects/RJRalsa/test.py", line 9, in <module>
    print(rtmidi.get_output_names())
  File "/usr/lib/python3/dist-packages/mido/backends/backend.py", line 175, in get_output_names
    devices = self._get_devices(**self._add_api(kwargs))
  File "/usr/lib/python3/dist-packages/mido/backends/backend.py", line 163, in _get_devices
    return self.module.get_devices(**self._add_api(kwargs))
  File "/usr/lib/python3/dist-packages/mido/backends/rtmidi.py", line 52, in get_devices
    input_names = set(rtmidi.MidiIn(rtapi=rtapi).get_ports())
  File "src/_rtmidi.pyx", line 817, in rtmidi._rtmidi.MidiIn.__cinit__
rtmidi._rtmidi.SystemError: MidiInAlsa::initialize: error creating ALSA sequencer client object.
```

The problem seems to be that it doesn't close connections to alsa:

```
...
l-wx------ 1 folkert folkert 64 aug  6 15:05 46 -> /dev/snd/seq
lrwx------ 1 folkert folkert 64 aug  6 15:05 47 -> /dev/snd/seq
lr-x------ 1 folkert folkert 64 aug  6 15:05 48 -> 'pipe:[592725]'
l-wx------ 1 folkert folkert 64 aug  6 15:05 49 -> 'pipe:[592725]'
l-wx------ 1 folkert folkert 64 aug  6 15:05 5 -> 'pipe:[592674]'
l-wx------ 1 folkert folkert 64 aug  6 15:05 50 -> /dev/snd/seq
```

**Comment by folkertvanheusden** [comment]
I don't believe this problem is in python-rtmidi as this runs flawlessly without leaking filedescriptors:

```
import rtmidi

midiout = rtmidi.MidiOut()

while True:
    available_ports = midiout.get_ports()
    print(available_ports)
```

**Comment by folkertvanheusden** [comment]
Ah, this was already fixed in 74cb885dbadc4754aabbee8001bf0d31b98b3029. Apparently ubuntu is a bit behind.
